frontend/src/pages/train/[train_space_id].tsx

```diff
@@ -11,272 +13,177 @@ import { Data, XAxisName, YAxisName } from "plotly.js";
+            // yaxis: { title: "Y axis" },
+            title: metric.name,
+            showlegend: true,
+            paper_bgcolor: "rgba(0,0,0,0)",
+            plot_bgcolor: "rgba(0,0,0,0)",
+          }}
+          config={{ responsive: true }}
+        />
+      );
+    } else if (metric.chart_type === "AUC/ROC") {
+      charts.push(
+        <Plot
+          data={[
+            {
+              name: "baseline",
+              x: [0, 1],
+              y: [0, 1],
+              type: "scatter",
+              marker: { color: "grey" },
+              line: {
+                dash: "dash",
+              },
+            },
+            ...(metric.values.map((x) => ({
+              name: `(AUC: ${x[2]})`,
+              x: x[0] as number[],
+              y: x[1] as number[],
+              type: "scatter",
+            })) as Data[]),
+          ]}
+          layout={{
+            height: 350,
+            width: 525,
+            xaxis: { title: "False Positive Rate" },
+            yaxis: { title: "True Positive Rate" },
+            title: "AUC/ROC Curves for your Deep Learning Model",
+            showlegend: true,
+            paper_bgcolor: "rgba(0,0,0,0)",
+            plot_bgcolor: "rgba(0,0,0,0)",
+          }}
+          config={{ responsive: true }}
+        />
+      );
+    } else if (metric.chart_type === "CONFUSION_MATRIX") {
+      charts.push(
+        <Plot
+          data={[
+            {
+              z: metric.values,
+              type: "heatmap",
+              colorscale: [
+                [0, "#e6f6fe"],
+                [1, "#003058"],
+              ],
+            },
+          ]}
+          layout={{
+            height: 525,
+            width: 525,
+            title: "Confusion Matrix (Last Epoch)",
+            xaxis: {
+              title: "Predicted",
+            },
+            yaxis: {
+              title: "Actual",
+              autorange: "reversed",
+            },
+            showlegend: true,
+            annotations: metric.values
+              .map((row, i) =>
+                row.map((_, j) => ({
+                  xref: "x1" as XAxisName,
+                  yref: "y1" as YAxisName,
+                  x: j,
+                  y: (i + metric.values.length - 1) % metric.values.length,
+                  text: metric.values[
+                    (i + metric.values.length - 1) % metric.values.length
+                  ][j].toString(),
+                  font: {
+                    color:
+                      metric.values[
+                        (i + metric.values.length - 1) % metric.values.length
+                      ][j] > 0
+                        ? "white"
+                        : "black",
+                  },
+                  showarrow: false,
+                }))
+              )
+              .flat(),
+            paper_bgcolor: "rgba(0,0,0,0)",
+            plot_bgcolor: "rgba(0,0,0,0)",
+          }}
+        />
+      );
```

Comment:
Questions:

1. Could we potentially separate these behaviors for different chart types into its own dedicated helper functions or components for better readability of `mapTrainResultsDataToCharts`
2. Didn't quite follow how the `metric.chart_type` parameter is configured to ensure the right charts appear for classification vs. regression problem type?

Reply:
1. Yes
2. It's more important to know what type of chart it is (line vs auc/roc) rather than if it's regression or classification, because both can have line charts

Reply:
got it. could we try to incorporate suggestion #1?

Reply:
done